src/UIAutomation/Microsoft.UI.UIAutomation/AutomationRemoteOperation.cpp

```diff
@@ -152,6 +152,32 @@ namespace winrt::Microsoft::UI::UIAutomation::implementation
         return make<AutomationRemoteAnyObject>(newId, *this);
     }
 
+    // NewEmpty is similar to NewNull except it doesn't insert any instruction, instead NewEmpty serves as a place holder
+    // for any operands that are to be set when the Remote Operation gets executed(an out only parameter without wasting an instruction)
+    // This is particularly used with CallExtension as it takes an arrray of operands and some of them will be
+    // out params and using NullOperand for all of them would waste instructions.
+    winrt::AutomationRemoteAnyObject AutomationRemoteOperation::NewEmpty()

```

Comment:
Could you add an issue to plumb this stuff through to UiaOperationAbstraction? Things that stand out to me are:

- Make `UiaEmpty` type
- Make `UiaByteArray` type
- Support `CallExtension` on `UiaTextRange` and a new `UiaConnectionBoundObject` maybe

Reply:
Certainly UiaConnectionBoundObject is needed. 
At the moment the general pattern for custom patterns is that you fetch a pattern from a UIAElement with CallExtension, and then call a method on that pattern again with CallExtension.
As the abstraction API currently has no UiaConnectionBoundObject, the caller must fetch the pattern as a UIAElement. (At least that is what I am doing in my real-world code right now). 
E.g. Given a UiaElement named 'element':
```
UiaElement pattern{nullptr};
element.CallExtension(somePatternGuid, pattern);
pattern.CallExtension(someMethodGuid, args...);
```
The pattern variable here should preferably be a UiaConnectionBoundObject, not UiaElement.
It would be nice if UiaElement and UiaTextRange could actually inherit from UiaConnectionBoundObject, and CallExtension / IsExtensionSupported be moved onto that UiaConnectionBoundObject, but I'm not quite sure that the magic Uia abstraction types could handle using inheritance like that...

Reply:
Created a new issue- https://github.com/microsoft/Microsoft-UI-UIAutomation/issues/93